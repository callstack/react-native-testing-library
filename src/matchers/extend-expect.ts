import { toBeOnTheScreen } from './to-be-on-the-screen';
import { toBeChecked } from './to-be-checked';
import { toBeCollapsed } from './to-be-collapsed';
import { toBeDisabled, toBeEnabled } from './to-be-disabled';
import { toBeBusy } from './to-be-busy';
import { toBeEmptyElement } from './to-be-empty-element';
import { toBeExpanded } from './to-be-expanded';
import { toBePartiallyChecked } from './to-be-partially-checked';
import { toBeSelected } from './to-be-selected';
import { toBeVisible } from './to-be-visible';
import { toContainElement } from './to-contain-element';
import { toHaveAccessibilityValue } from './to-have-accessibility-value';
import { toHaveAccessibleName } from './to-have-accessible-name';
import { toHaveDisplayValue } from './to-have-display-value';
import { toHaveProp } from './to-have-prop';
import { toHaveStyle } from './to-have-style';
import { toHaveTextContent } from './to-have-text-content';

export type * from './types';

expect.extend({
  toBeOnTheScreen,
  toBeChecked,
  toBeCollapsed,
  toBeDisabled,
  toBeBusy,
  toBeEmptyElement,
  toBeEnabled,
  toBeExpanded,
  toBePartiallyChecked,
  toBeSelected,
  toBeVisible,
  toContainElement,
  toHaveAccessibilityValue,
  toHaveAccessibleName,
  toHaveDisplayValue,
  toHaveProp,
  toHaveStyle,
  toHaveTextContent,
});
