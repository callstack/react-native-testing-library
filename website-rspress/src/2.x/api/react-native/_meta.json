[
  {
    "type": "file",
    "name": "index",
    "label": "Table of Contents"
  },
  {
    "type": "dir",
    "name": "classes",
    "label": "Classes"
  },
  {
    "type": "dir",
    "name": "interfaces",
    "label": "Interfaces"
  },
  {
    "type": "dir",
    "name": "types",
    "label": "Types"
  }
]
