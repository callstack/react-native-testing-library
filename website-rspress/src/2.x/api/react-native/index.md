# Re.Pack React Native APIs

## Table of contents

### Classes

- [Chunk](./classes/Chunk.md)
- [ChunkManager](./classes/ChunkManager.md)

### Interfaces

- [ChunkConfig](./interfaces/ChunkConfig.md)
- [ChunkManagerConfig](./interfaces/ChunkManagerConfig.md)
- [RemoteChunkLocation](./interfaces/RemoteChunkLocation.md)
- [StorageApi](./interfaces/StorageApi.md)

### Type aliases

- [RemoteChunkResolver](./types/RemoteChunkResolver.md)
