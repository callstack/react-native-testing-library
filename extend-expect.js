require('./build/matchers/extend-expect');
