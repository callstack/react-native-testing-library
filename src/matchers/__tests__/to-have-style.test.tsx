import React from 'react';
import { View, Text, StyleSheet, Pressable } from 'react-native';
import { render } from '../..';
import '../extend-expect';

test('handles positive test cases', () => {
  const styles = StyleSheet.create({
    container: { borderBottomColor: 'white' },
  });
  const { getByTestId } = render(
    <View
      testID="container"
      style={[
        {
          backgroundColor: 'blue',
          height: '40%',
          transform: [{ scale: 2 }, { rotate: '45deg' }],
        },
        [{ height: '100%' }],
        [[{ width: '50%' }]],
        styles.container,
      ]}
    >
      <Text>Hello World</Text>
    </View>
  );

  const container = getByTestId('container');

  expect(container).toHaveStyle({ backgroundColor: 'blue', height: '100%' });
  expect(container).toHaveStyle([
    { backgroundColor: 'blue' },
    { height: '100%' },
  ]);
  expect(container).toHaveStyle({ backgroundColor: 'blue' });
  expect(container).toHaveStyle({ height: '100%' });
  expect(container).toHaveStyle({ borderBottomColor: 'white' });
  expect(container).toHaveStyle({ width: '50%' });
  expect(container).toHaveStyle([[{ width: '50%' }]]);
  expect(container).toHaveStyle({
    transform: [{ scale: 2 }, { rotate: '45deg' }],
  });
});

test('handles negative test cases', () => {
  const { getByTestId } = render(
    <View
      testID="container"
      style={{
        backgroundColor: 'blue',
        borderBottomColor: 'black',
        height: '100%',
        transform: [{ scale: 2 }, { rotate: '45deg' }],
      }}
    >
      <Text>Hello World</Text>
    </View>
  );

  const container = getByTestId('container');
  expect(() =>
    expect(container).toHaveStyle({
      backgroundColor: 'blue',
      transform: [{ scale: 1 }],
    })
  ).toThrowErrorMatchingSnapshot();
  expect(container).not.toHaveStyle({ fontWeight: 'bold' });
  expect(container).not.toHaveStyle({ color: 'black' });
  expect(container).not.toHaveStyle({
    transform: [{ rotate: '45deg' }, { scale: 2 }],
  });
  expect(container).not.toHaveStyle({ transform: [{ rotate: '45deg' }] });
});

test('handles when the style prop is undefined', () => {
  const { getByTestId } = render(
    <View testID="container">
      <Text>Hello World</Text>
    </View>
  );

  const container = getByTestId('container');

  expect(container).not.toHaveStyle({ fontWeight: 'bold' });
});

test('handles transform when transform undefined', () => {
  const { getByTestId } = render(
    <View
      testID="container"
      style={{
        backgroundColor: 'blue',
        transform: undefined,
      }}
    >
      <Text>Hello World</Text>
    </View>
  );

  const container = getByTestId('container');
  expect(() =>
    expect(container).toHaveStyle({ transform: [{ scale: 1 }] })
  ).toThrowErrorMatchingSnapshot();
});

test('handles Pressable with function style prop', () => {
  const { getByTestId } = render(
    <Pressable testID="test" style={() => ({ backgroundColor: 'blue' })} />
  );
  expect(getByTestId('test')).toHaveStyle({ backgroundColor: 'blue' });
});
