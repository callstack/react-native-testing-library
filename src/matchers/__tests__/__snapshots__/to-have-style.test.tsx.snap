// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles negative test cases 1`] = `
"expect(element).toHaveStyle()

- Expected

  backgroundColor: blue;
  transform: [
    {
-     "scale": 1
+     "scale": 2
+   },
+   {
+     "rotate": "45deg"
    }
  ];"
`;

exports[`handles transform when transform undefined 1`] = `
"expect(element).toHaveStyle()

- Expected

- transform: [
-   {
-     "scale": 1
-   }
- ];
+ transform: undefined;"
`;
